ttg/ttg/util/meta.h

```diff
@@ -83,6 +83,69 @@ namespace ttg {
     template <typename Tuple, template <typename> typename Pred>
     using filtered_tuple_t = typename filtered_tuple<Tuple, Pred>::type;
 
+    template <template <typename> typename Predicate, bool Default, typename... Ts>
```

Comment:
Can you add some documentation on what `probe_first` and `probe_last` do?

Reply:
done